lib/meta/mutations.go

```diff
@@ -0,0 +1,140 @@
+package meta
```

Comment:
I wonder if we should consider using JSON patch rather than creating our own interfaces for mutations?

Reply:
interesting idea. I will try it out and let you know how it does.